### PR TITLE
[filmmodu] Add new extractor

### LINKED ISSUE #288: Add support for filmmodu.org
## Checklist

- [x] I'm reporting a new site support request
- [x] I've verified that I'm running yt-dlp version **2021.04.22**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [ ] I've checked that none of provided URLs violate any copyrights
- [x] I've searched the bugtracker for similar site support requests including closed ones

## Example URLs

- https://www.filmmodu.org/avengers-endgame-hd-turkce-dublaj-izle
- https://www.filmmodu.org/avengers-endgame-hd-altyazili-izle

## Description

- Sample page URL https://www.filmmodu.org/traffic-turkce-dublaj-izle
- Extract movie_id form page source e.g. var videoId = '1585'
- Extract videoType form page source e.g. var videoType = 'tr'
- Create m3u8 url like https://www.filmmodu.org/get-source?movie_id=videoId&type=videoType
- final URL should like https://www.filmmodu.org/get-source?movie_id=1585&type=tr

### PR DESCRIPTION
**Pull request by mzbaulhaque**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Adding new extractor to download movies from [filmmodu.org](https://www.filmmodu.org/).
Should fix #288.

### CODE REVIEW

File: yt_dlp/extractor/filmmodu.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        formats = [
+            {
+                'url': source['src'],
+                'format_id': source['label'],
+                'resolution': '{}p'.format(source['res']),
+                'width': int_or_none(source['res']),
+            }
+            for source in data['sources']
+        ]
+
```
Comment (pukkandan) [inline yt_dlp/extractor/filmmodu.py:58]:
```suggestion
        formats = [{
            'url': source['src'],
            'format_id': source.get('label'),
            'width': int_or_none(source.get('res')),
        } for source in data['sources']]

```

File: yt_dlp/extractor/filmmodu.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        subtitles = {}
+
+        if data.get('subtitle'):
+            subtitles['tr'] = [{
+                'url': data['subtitle'],
+            }]
+
+        return {
+            'id': real_video_id,
+            'title': title,
```
Comment (pukkandan) [inline yt_dlp/extractor/filmmodu.py:70]:
```suggestion
            'display_id': video_id,
            'title': title,
```

File: yt_dlp/extractor/filmmodu.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+
+class FilmmoduIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www.)?filmmodu.org/(?P<id>[^/]+-(?:turkce-dublaj-izle|altyazili-izle))'
+    _TESTS = [{
+        'url': 'https://www.filmmodu.org/f9-altyazili-izle',
+        'md5': '26eb61d8bab018b9189ded692c352a53',
+        'info_dict': {
+            'id': '10804',
+            'ext': 'm3u8',
```
Comment (pukkandan) [inline yt_dlp/extractor/filmmodu.py:15]: The video's ext is clearly not `m3u8`. This should be set correctly

File: yt_dlp/extractor/filmmodu.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        real_video_id = self._search_regex(r'var\s*videoId\s*=\s*\'([0-9]+)\'', webpage, 'video_id')
+        video_type = self._search_regex(r'var\s*videoType\s*=\s*\'([a-z]+)\'', webpage, 'video_type')
+        data = self._download_json('https://www.filmmodu.org/get-source', real_video_id, query={
+            'movie_id': real_video_id,
+            'type': video_type,
+        })
+        formats = [{
+            'url': source['src'],
+            'format_id': source.get('label'),
+            'width': int_or_none(source.get('res')),
```
Comment (pukkandan) [inline yt_dlp/extractor/filmmodu.py:52]: are you sure this is width and not height?

Comment (mzbaulhaque) [reply yt_dlp/extractor/filmmodu.py]: yes it's height, put width by mistake

**Pull request merged**